switchtool/switch/switch.py

```diff
@@ -94,36 +101,32 @@ class Switch(netconfig.host.Host):
     _vlan = []
     _user = "admin"
 
-    def __init__(self, switch_name, user="admin", pw=None, enablepw=None, type=None):
-        self._nc = netconfig.NetConfig()
-
-        if not type:
-            type = determine_type(self._nc, switch_name)
+    def __init__(
+        self, switch_name, user="admin", pw=None, enablepw=None, switch_type=None
+    ):
+        if switch_type is None:
+            switch_type = determine_type(switch_name)
             module_logger.info(
-                "No switch type supplied guessing " "that switch is type {:}".format(
-                    type
+                "No switch type supplied guessing that switch is type {:}".format(
+                    switch_type
                 )
             )
 
-        self.type = type
-        host_info = list(self._nc.find_hosts(switch_name, as_dict=True).values())[0]
-        super(Switch, self).__init__(switch_name, host_info)
+        self.name = switch_name
+        self.switch_type = switch_type
 
         # Check host
-        ping = self.ping()
-        if not ping:
+        if not ping(switch_name):
             raise IOError("Unable to ping {:}".format(switch_name))
-        # Request password if not supplied
-        if user:
-            self._user = user
 
+        self._user = user
         self._pw = pw
         self._enablepw = enablepw
 
     @property
     def subnets(self):
         """
-        The list of vlan number,subnet pairs that are found on the switch
+        The list of vlan number, subnet pairs that are found on the switch
```

Comment:
```suggestion
        The list of (vlan number, subnet) pairs that are found on the switch
```

Maybe it's only my brain that thought there'd be pairs of subnets